diff --git a/ament_clang_format/main.py b/ament_clang_format/main.py
index edfa485..740aa80 100755
--- a/ament_clang_format/main.py
+++ b/ament_clang_format/main.py
@@ -18,4 +18,5 @@ import argparse
 import argparse
 import os
+import shutil
 import subprocess
 import sys
@@ -249,8 +250,6 @@ def find_executable(file_names):
 def find_executable(file_names):
-    paths = os.getenv('PATH').split(os.path.pathsep)
     for file_name in file_names:
-        for path in paths:
-            file_path = os.path.join(path, file_name)
-            if os.path.isfile(file_path) and os.access(file_path, os.X_OK):
-                return file_path
+        file_path = shutil.which(file_name)
+        if file_path:
+            return file_path
     return None
diff --git a/ament_clang_tidy/main.py b/ament_clang_tidy/main.py
index 7ad2820..223c6c5 100755
--- a/ament_clang_tidy/main.py
+++ b/ament_clang_tidy/main.py
@@ -24,4 +24,5 @@ import json
 import os
 import re
+import shutil
 import subprocess
 import sys
@@ -269,8 +270,6 @@ def find_executable(file_names):
 def find_executable(file_names):
-    paths = os.getenv('PATH').split(os.path.pathsep)
     for file_name in file_names:
-        for path in paths:
-            file_path = os.path.join(path, file_name)
-            if os.path.isfile(file_path) and os.access(file_path, os.X_OK):
-                return file_path
+        file_path = shutil.which(file_name)
+        if file_path:
+            return file_path
     return None
